statsd/statsd.go

```diff
@@ -133,10 +135,43 @@ func (c *Client) append(cmd string) error {
 	return nil
 }
 
+func joinMaxSize(a []string, sep string, maxSize int) []string {
+	if len(a) == 0 {
+		return []string{""}
+	}
+	if len(a) == 1 {
+		return a
+	}
+	n := len(sep) * (len(a) - 1)
+	for i := 0; i < len(a); i++ {
+		n += len(a[i])
+	}
+	splitSize := (n / maxSize) + 1
+	lastIndex := 0
+	container := make([]string, splitSize)
```

Comment:
The calculation of splitSize to me seems both unnecessary and wrong.  It's unnecessary because what you're actually doing here is creating a list of a bunch of empty strings, and then adding more strings to the end after them.  You're not actually overwriting those strings that you put there!

And I think the calculation is wrong too... Imagine you have, say, 5 messages of length 800.  They will all need to go in their own message (so 5 messages) because 800 + 800 > 1432. But this math will decide we need (5 * 800 / 1432) + 1 = 3.79 messages.

But the fix is simple: just do `var container []string` instead of doing any of the calculation here.